README.md

```diff
@@ -1,91 +1,157 @@
+
+## Repository Structure
+
+```plaintext
+.
+├── .github/                    # GitHub Actions workflows (CI/CD, not for workflow-function logic)
+├── manifest.json               # Plugin manifest for Ubiquity OS Kernel
+├── supabase/                   # SQL migration files for Supabase schema
+├── src/                        # Source code
+│   ├── adapters/               # Storage adapters (e.g., Supabase integrations)
+│   ├── bot/                    # Core Telegram bot functionality (Worker and Workflow)
+│   │   ├── features/           # Bot features, including commands and event handlers
+│   │   ├── filters/            # Grammy filters (e.g., isAdmin, isPrivateChat)
+│   │   ├── handlers/           # Stateful command handlers based on chat types or user privileges
+│   │   ├── helpers/            # Utility functions like logging and custom Grammy contexts
+│   │   ├── middlewares/        # Middleware functions (e.g., rate limiting, session management)
+│   │   ├── mtproto-api/        # MTProto API client for advanced Telegram features
+│   │   │   ├── bot/            # MTProto API session management, 2FA auth scripts, Telegram client setup
+│   │   │   ├── workrooms/      # A collection of Workflow-functions leveraging MTProto API
+│   │   ├── index/              # Bot initialization, where commands and features are registered
+│   ├── handlers/               # General plugin handlers (e.g., GitHub API functions, webhook processing)
+│   ├── server/                 # Hono app for managing Cloudflare Worker instance of the bot
+│   ├── types/                  # Typebox schemas, TypeScript types, Singleton classes
+│   ├── utils/                  # General utility functions and helpers
+│   ├── worker.ts               # Main entry point, routes events (GitHub or Telegram) to appropriate handlers
+│   ├── plugin.ts               # GitHub event handler, forwards events to `workflow-entry.ts` or processes them
+│   ├── workflow-entry.ts       # Handles forwarded GitHub events, MTProto API interactions, and workflow logic
+```
 
-- [Full Ubiquibot Configuration](https://github.com/ubiquity/ubiquibot/blob/0fde7551585499b1e0618ec8ea5e826f11271c9c/src/types/configuration-types.ts#L62) - helpful for defining your plugin's settings as they are strongly typed and will be validated by the kernel.
-- [Ubiquibot V1](https://github.com/ubiquity/ubiquibot) - helpful for porting V1 functionality to V2, helper/utility functions, types, etc. Everything is based on the V1 codebase but with a more modular approach. When using V1 code, keep in mind that most all code will need refactored to work with the new V2 architecture.
+## Considerations
 
-## Examples
+The bot has two "branches" due to Cloudflare Workers' limitations in handling long-running processes. The **Worker** handles most bot commands and API interactions, while the **Workflow** manages features the bot cannot perform.
```

Comment:
Pretty interesting and wild architecture where we're able to procure the benefits from both workers and actions. 

Maybe this is the future of plug-in development